File: AzardBackend/src/main/java/com/azard/service/WasteService.java
package com.azard.service;

import java.io.IOException;
import java.security.GeneralSecurityException;
import java.sql.SQLException;

import org.springframework.web.multipart.MultipartFile;

import com.azard.model.Waste;
import com.azard.model.WasteReport;

public interface WasteService {

	WasteReport searchWastes(Long startDateMilliseconds, Long endDateMilliseconds, String storeIds, String productCode,
			Integer deviceBrandId, Integer deviceModelId);

	void submitWaste(Waste waste, MultipartFile wasteImage) throws SQLException, IllegalStateException, IOException, GeneralSecurityException;

}

File: AzardFrontend/app/scripts/services/WasteService.js
'use strict';

angular.module('kalafcheFrontendApp')
	.service('WasteService', function($http, Environment) {
		angular.extend(this, {
			submitWaste: submitWaste,
			searchWastes: searchWastes
		});

        function submitWaste(waste, file) {
            var fileFormData = new FormData();
            fileFormData.append('wasteImage', file);
            fileFormData.append('itemId', waste.itemId);
            fileFormData.append('description', waste.description);

            return $http.post(Environment.apiEndpoint + '/AzardBackend/waste', fileFormData, {
                transformRequest: angular.identity,
                headers: {'Content-Type': undefined}
 
            }).then(function (response) {
                return response.data;
            })
        }

        function searchWastes(startDateMilliseconds, endDateMilliseconds, storeIds,
        	selectedBrandId, selectedModelId, productCode) { 
            var params = {"params" : 
            	{"startDateMilliseconds": startDateMilliseconds, 
            	"endDateMilliseconds": endDateMilliseconds, 
                "storeIds": storeIds, 
                "deviceBrandId": selectedBrandId, 
                "deviceModelId": selectedModelId, 
                "productCode": productCode}};
            console.log(params);

            return $http.get(Environment.apiEndpoint + '/AzardBackend/waste', params)
                .then(
                    function(response) {
                        console.log(response.data);
                        return response.data
                    }
                );
        }

	});